hazelcast/src/test/java/com/hazelcast/jet/JobStatusListenerTest.java

```diff
@@ -383,81 +359,63 @@ public void jobStatusChanged(JobStatusEvent e) {
                     threadInfo));
         }
 
-        /**
-         * {@link JetTestSupport#assertJobStatusEventually assertJobStatusEventually(job, RUNNING)}
-         * is not reliable to wait for RESTART since the initial status is RUNNING. For such cases,
-         * {@link HazelcastTestSupport#assertEqualsEventually assertEqualsEventually(listener::runCount,
-         * lastRunCount + 1)} can be used.
-         */
-        public int runCount() {
-            return (int) instance.get().getMap(RUN_MAP_NAME).getOrDefault(job.getId(), 0);
-        }
-
-        public void advance(int count) {
-            JobStatusListenerTest.this.advance(job.getId(), count);
-        }
-
         public void deregister() {
             job.removeStatusListener(registrationId);
         }
     }
 
-    protected void advance(long jobId, int count) {
-        instance.get().getMap(ADVANCE_MAP_NAME).put(jobId, count);
-    }
+    protected class TestCase {
+        private final JobConfig config;
+        private final DAG dag;
+        private BiConsumerEx<Job, JobStatusLogger> test;
+        private String[] log;
 
-    protected static BatchSource<Integer> batchSource() {
-        return batchSource(r -> { });
-    }
+        public TestCase(JobConfig config, DAG dag) {
```

Comment:
you could be even more fluent and have methods like `withConfig`, `executingJob(Dag)` etc.